src/upgrade/sender.rs

```diff
@@ -1,28 +1,26 @@
-use async_std::sync;
-
 use crate::upgrade::Connection;
 
 /// The sending half of a channel to send an upgraded connection.
 ///
-/// Unlike `async_std::sync::channel` the `send` method on this type can only be
+/// Unlike `async_channel::Sender` the `send` method on this type can only be
 /// called once, and cannot be cloned. That's because only a single instance of
 /// `Connection` should be created.
 #[derive(Debug)]
 pub struct Sender {
-    sender: sync::Sender<Connection>,
+    sender: async_channel::Sender<Connection>,
 }
 
 impl Sender {
     /// Create a new instance of `Sender`.
     #[doc(hidden)]
-    pub fn new(sender: sync::Sender<Connection>) -> Self {
+    pub fn new(sender: async_channel::Sender<Connection>) -> Self {
         Self { sender }
     }
 
     /// Send a `Trailer`.
     ///
     /// The channel will be consumed after having sent trailers.
     pub async fn send(self, trailers: Connection) {
-        self.sender.send(trailers).await
+        let _ = self.sender.send(trailers).await;
```

Comment:
This line seems worth discussing. async-std channels cannot be closed as long as there's a sender, but async-channels can be, and return an Err variant if the channel is closed. Probably the best fix for this is to return that Result, making this semver-major. As written, this silently swallows the possible SendError.

Reply:
Functionally, this is the same as existing behavior. It's a oneshot channel where the sender doesn't care if the receiver is dropped. It might be a good idea to return a `Result` here if the receiver has been dropped, but I see it as a separate issue.

Reply:
Since this `Sender`/`Receiver` code is duplicated a couple times, I'd recommend making a generic `oneshot` submodule that's shared by `trailers`, `upgrade`, and whatever else relies on this pattern.

Reply:
I was planning that as a follow-up, but I wanted to keep this PR non-breaking.